### ISSUE TITLE
[Bug] Local Connection (in addition to Cloud) Error Won't connect

### ISSUE THREAD

**Issue by spikked27** [greghesp/ha-bambulab#598]
### Describe the bug

Can't locally connect to P1S when setting up with Bambu Cloud, keep receiving error to check the IP address.

### To Reproduce

1. Configure new printer with bambu cloud
2. Input login credentials
3. Input local IP address and access code for local access

### Expected Behaviour

Expected that local connection would work, relevant debug logs are below capturing a few failed attempts followed by successful cloud setup without the local IP address i.e. no camera access.

### What device are you using?

P1S

### Diagnostic Output

```shell
Diagnostic log output below with redacted login credentials
```

### Log Extracts

```shell
2024-10-04 09:28:52.272 DEBUG (SyncWorker_43) [custom_components.bambu_lab.pybambu] Getting accessToken from Bambu Cloud
2024-10-04 09:28:52.272 DEBUG (SyncWorker_43) [custom_components.bambu_lab.pybambu] Data = {'account': 'XXXXXXXXXX', 'password': 'XXXXXXXXXXX'}
2024-10-04 09:28:52.487 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Choose_Device
2024-10-04 09:28:52.487 DEBUG (SyncWorker_28) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:28:52.649 DEBUG (MainThread) [custom_components.bambu_lab] Printer count = <built-in method count of list object at 0x7fb69bd7a480>
2024-10-04 09:28:53.984 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Choose_Device
2024-10-04 09:28:53.984 DEBUG (SyncWorker_0) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:28:54.131 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Lan
2024-10-04 09:28:54.131 DEBUG (SyncWorker_33) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:29:21.324 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Lan
2024-10-04 09:29:21.325 DEBUG (SyncWorker_53) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:29:21.465 DEBUG (MainThread) [custom_components.bambu_lab] Config Flow: Testing local mqtt connection
2024-10-04 09:29:21.465 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Try Connection
2024-10-04 09:29:21.480 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Test connection: Connecting to 192.168.0.157
2024-10-04 09:29:26.486 DEBUG (MainThread) [custom_components.bambu_lab.pybambu]  Disconnect: Client Disconnecting
2024-10-04 09:29:28.110 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Lan
2024-10-04 09:29:28.110 DEBUG (SyncWorker_62) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:29:28.244 DEBUG (MainThread) [custom_components.bambu_lab] Config Flow: Testing local mqtt connection
2024-10-04 09:29:28.245 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Try Connection
2024-10-04 09:29:28.259 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Test connection: Connecting to 192.168.0.157
2024-10-04 09:29:33.265 DEBUG (MainThread) [custom_components.bambu_lab.pybambu]  Disconnect: Client Disconnecting
2024-10-04 09:29:44.329 DEBUG (SyncWorker_2) [custom_components.bambu_lab.pybambu] Getting accessToken from Bambu Cloud
2024-10-04 09:29:44.329 DEBUG (SyncWorker_2) [custom_components.bambu_lab.pybambu] Data = {'account': 'XXXXXXXXXXXX', 'password': 'XXXXXXXXXXXXXX'}
2024-10-04 09:29:44.516 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Choose_Device
2024-10-04 09:29:44.517 DEBUG (SyncWorker_31) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:29:44.653 DEBUG (MainThread) [custom_components.bambu_lab] Printer count = <built-in method count of list object at 0x7fb6858e4c00>
2024-10-04 09:29:45.593 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Choose_Device
2024-10-04 09:29:45.593 DEBUG (SyncWorker_3) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:29:45.756 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Lan
2024-10-04 09:29:45.756 DEBUG (SyncWorker_39) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:29:49.072 DEBUG (MainThread) [custom_components.bambu_lab] async_step_Bambu_Lan
2024-10-04 09:29:49.073 DEBUG (SyncWorker_30) [custom_components.bambu_lab.pybambu] Getting device list from Bambu Cloud
2024-10-04 09:29:49.226 DEBUG (MainThread) [custom_components.bambu_lab] Config Flow: Writing entry: 'Bambu P1S'
2024-10-04 09:29:49.226 DEBUG (MainThread) [custom_components.bambu_lab] async_setup_entry Start
2024-10-04 09:29:49.226 DEBUG (MainThread) [custom_components.bambu_lab] ConfigEntry.Id: XXXXXXXXXXXXXXX
2024-10-04 09:29:49.227 DEBUG (MainThread) [custom_components.bambu_lab] P1S HA POLL: MQTT connected: False
2024-10-04 09:29:49.227 DEBUG (MainThread) [custom_components.bambu_lab] Finished fetching bambu_lab data in 0.000 seconds (success: True)
2024-10-04 09:29:49.233 DEBUG (MainThread) [custom_components.bambu_lab] BUTTON::async_setup_entry
2024-10-04 09:29:49.236 DEBUG (MainThread) [custom_components.bambu_lab] CAMERA::async_setup_entry
2024-10-04 09:29:49.236 DEBUG (MainThread) [custom_components.bambu_lab] FAN::async_setup_entry
```

### Other Information

Log Outputs from HA:

This error originated from a custom integration.

Logger: custom_components.bambu_lab.pybambu
Source: custom_components/bambu_lab/pybambu/bambu_client.py:397
integration: Bambu Lab (documentation, issues)
First occurred: 4:00:53 AM (4 occurrences)
Last logged: 9:23:16 AM

On Disconnect: Disconnected from Broker: 16
On Disconnect: Disconnected from Broker: 0

This error originated from a custom integration.

Logger: custom_components.bambu_lab.pybambu
Source: custom_components/bambu_lab/pybambu/bambu_client.py:401
integration: Bambu Lab (documentation, issues)
First occurred: October 3, 2024 at 10:43:39 PM (12 occurrences)
Last logged: 9:23:16 AM

_on_disconnect

This error originated from a custom integration.

Logger: custom_components.bambu_lab
Source: custom_components/bambu_lab/config_flow.py:124
integration: Bambu Lab (documentation, issues)
First occurred: 9:25:17 AM (1 occurrences)
Last logged: 9:25:17 AM

Failed to connect with error code (400,)

**Comment by coolnetworks** [comment]
FYI this wont connect local only.

**Comment by AdrianGarside** [comment]
Do you have MFA enabled for your bambu account?

**Comment by beast3334** [comment]
Seeing the same issue and logs. Can't setup new A1 Mini over LAN, logs only show attempting to connect and then disconnecting.

**Comment by AdrianGarside** [comment]
> Seeing the same issue and logs. Can't setup new A1 Mini over LAN, logs only show attempting to connect and then disconnecting.

Need more information. What version of the integration re you running and please provide debug logs.

**Comment by beast3334** [comment]
I'm using v2.0.24, the following is what I'm getting logged out while having Debugging Logging enabled (anonymized my serial number).

```
2024-10-23 12:24:32.387 DEBUG (MainThread) [custom_components.bambu_lab] Config Flow: Testing local mqtt connection
2024-10-23 12:24:32.387 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Try Connection
2024-10-23 12:24:32.415 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Test connection: Connecting to 192.168.251.158
2024-10-23 12:24:33.278 INFO (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Connected to printer
2024-10-23 12:24:33.278 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Now test subscribing...
2024-10-23 12:24:33.278 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Subscribing: device/**********/report
2024-10-23 12:24:33.278 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Getting version info
2024-10-23 12:24:33.278 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Sent {'info': {'sequence_id': '0', 'command': 'get_version'}} to topic device/**********/request
2024-10-23 12:24:33.279 WARNING (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Disconnect: Printer disconnected with error code: 5
2024-10-23 12:24:33.279 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] _on_disconnect: Lost connection to the printer
2024-10-23 12:24:35.122 INFO (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Connected to printer
2024-10-23 12:24:35.122 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Now test subscribing...
2024-10-23 12:24:35.122 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Subscribing: device/**********/report
2024-10-23 12:24:35.123 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Getting version info
2024-10-23 12:24:35.124 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Sent {'info': {'sequence_id': '0', 'command': 'get_version'}} to topic device/***********/request
2024-10-23 12:24:35.124 WARNING (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Disconnect: Printer disconnected with error code: 5
2024-10-23 12:24:35.124 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] _on_disconnect: Lost connection to the printer
2024-10-23 12:24:35.902 DEBUG (P1S-Mqtt-148) [custom_components.bambu_lab.pybambu] event_printer_data_update
2024-10-23 12:24:37.938 DEBUG (P1S-Mqtt-148) [custom_components.bambu_lab.pybambu] event_printer_data_update
2024-10-23 12:24:37.992 INFO (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Connected to printer
2024-10-23 12:24:37.993 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Now test subscribing...
2024-10-23 12:24:37.993 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Subscribing: device/**********/report
2024-10-23 12:24:37.993 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Getting version info
2024-10-23 12:24:37.993 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Sent {'info': {'sequence_id': '0', 'command': 'get_version'}} to topic device/***********/request
2024-10-23 12:24:37.994 WARNING (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Disconnect: Printer disconnected with error code: 5
2024-10-23 12:24:37.994 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] _on_disconnect: Lost connection to the printer
2024-10-23 12:24:41.982 DEBUG (P1S-Mqtt-148) [custom_components.bambu_lab.pybambu] event_printer_data_update
2024-10-23 12:24:42.840 INFO (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Connected to printer
2024-10-23 12:24:42.840 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Now test subscribing...
2024-10-23 12:24:42.840 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Subscribing: device/**********/report
2024-10-23 12:24:42.840 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Connect: Getting version info
2024-10-23 12:24:42.840 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] Sent {'info': {'sequence_id': '0', 'command': 'get_version'}} to topic device/**********/request
2024-10-23 12:24:42.841 WARNING (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] On Disconnect: Printer disconnected with error code: 5
2024-10-23 12:24:42.841 DEBUG (Thread-10 (_thread_main)) [custom_components.bambu_lab.pybambu] _on_disconnect: Lost connection to the printer
2024-10-23 12:24:43.256 DEBUG (MainThread) [custom_components.bambu_lab.pybambu]  Disconnect: Client Disconnecting
```

**Comment by AdrianGarside** [comment]
What is your printer firmware version?

**Comment by InventasEirik** [comment]
I have the same issue, running 00.00.06.40

**Comment by AdrianGarside** [comment]
All the recent issues were due to cloudflare rollout at Bambu's end. Currently I believe those have all been mitigated.

@spikked27 are you still seeing connectivity issues with the latest version of the integation (2.0.25)

**Comment by nramo013** [comment]
Still having an issue with latest update for HA and Bambu Lab Add On, latest Firmware for Pronters. Using X1C on both printers

**Comment by AdrianGarside** [comment]
@nramo013 please provide debug logs covering the failed connection attempt.

**Comment by AdrianGarside** [comment]
@beast3334 that should have been fixed in 2.0.25

**Comment by AdrianGarside** [comment]
Closing this issue. Please create a new issue if you still have connectivity problems on 2.0.25+
